master/pkg/model/experiment.go

```diff
@@ -422,6 +440,49 @@ type Checkpoint struct {
 	DeterminedVersion string     `db:"determined_version" json:"determined_version"`
 }
 
+// CheckpointV2 represents a row from the `checkpoints_v2` table.
+type CheckpointV2 struct {
+	ID           int              `db:"id"`
+	UUID         uuid.UUID        `db:"uuid"`
+	TaskID       TaskID           `db:"task_id"`
+	AllocationID AllocationID     `db:"allocation_id"`
+	ReportTime   time.Time        `db:"report_time"`
+	State        State            `db:"state"`
+	Resources    map[string]int64 `db:"resources"`
+	Metadata     JSONObj          `db:"metadata"`
+}
+
+// CheckpointTrainingMetadata is a substruct of checkpoints encapsulating training specific
+// information.
+type CheckpointTrainingMetadata struct {
+	TrialID           int     `db:"trial_id"`
+	ExperimentID      int     `db:"experiment_id"`
+	ExperimentConfig  JSONObj `db:"experiment_config"`
+	HParams           JSONObj `db:"hparams"`
+	TrainingMetrics   JSONObj `db:"training_metrics"`
+	ValidationMetrics JSONObj `db:"validation_metrics"`
+	SearcherMetrics   JSONObj `db:"searcher_metric"`
```

Comment:
There is an inconsistency between this being a JSONObj and model.experiment considering it a float64. Which is it?